### PR TITLE
Add package-level comment and update docs

### PR DESCRIPTION
**Pull request by vladimir-ch**
~~Also panic if the method order is not 1 in Gradient. An alternative solution would be not calling the function Gradient.~~

### CODE REVIEW

**Comment by btracey** [comment]
Remove the panic from this. The behavior/semantic change is distinct from the documentation cleanups

**Comment by vladimir-ch** [comment]
Removed. PTAL @btracey

File: fd/diff.go [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package fd provides functions to compute finite difference approximations
```
Comment (btracey) [inline fd/diff.go]: to approximate derivatives using finite difference.

File: fd/diff.go [hunk 2]
```diff
@@ -12,19 +14,21 @@ import (
 	"github.com/gonum/floats"
 )
 
-// A Point is a stencil location in a difference method.
+// A Point is a stencil location in a finite difference formula.
```
Comment (btracey) [inline fd/diff.go]: scheme instead of formula?

File: fd/diff.go [hunk 3]
```diff
@@ -12,19 +14,21 @@ import (
...
-// Method is a specific finite difference method. Method specifies the stencil,
-// that is, the function locations (relative to x) which will be used to estimate
-// the derivative. It also specifies the order of derivative it estimates. Order = 1
-// represents the derivative, Order = 2 represents the curvature, etc.
+// Method represents a finite difference formula that approximates
+// the derivative of order k of a function f at x as
+//  d^k f(x) ≅ (1 / h^k) * \sum_i Coeff_i * f(x + h * Loc_i),
+// where h is a small positive step.
 type Method struct {
+	// Stencil is the set of sampling locations relative to x which are used to
```
Comment (btracey) [inline fd/diff.go]: Probably worth noting that the locations in the Stencil will be scaled by Step.

**Comment by btracey** [comment]
LGTM with changes.

File: fd/diff.go (hunk 2 above)
Comment (vladimir-ch) [inline fd/diff.go]: I don't like neither method nor scheme in this context, in my mind they are both too related to discretization of PDEs. I would prefer formula or kernel (or something else). I have a PR ready where I rename `Method` to `Formula`, I think that is a more accurate description. I will leave this as is and we can return to this later in that PR.

File: fd/diff.go (hunk 3 above)
Comment (vladimir-ch) [inline fd/diff.go]: I will add it.

File: fd/diff.go (hunk 2 above)
Comment (btracey) [inline fd/diff.go]: Sounds good.

**Pull request merged**